トップページの実装 DB設計

Daisekkei opened the pull request:
What
１、Chat spaceのDB設計に関して、
Group usersテーブルを追加
２、AssociationにBlongs _toメソッドを追加

Why
１、中間テーブルであるgroups_usersテーブル作成のため
２、該当ユーザー、グループメンバー以外コメントを編集できないようにするため

Daisekkei committed:
Update README.md

What
１、Chat spaceのDB設計に関して、
Group usersテーブルを追加
２、AssociationにBlongs _toメソッドを追加

Why
１、中間テーブルであるgroups_usersテーブル作成のため
２、該当ユーザー、グループメンバー以外コメントを編集できないようにするため

i-date commented on README.md:
```diff
@@ -22,3 +22,14 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
```
30-31行目の外部キーはinteger型ではなく、references型を使用しましょう。
そうでない場合、`foreign_key: true`では外部キー制約にならないので注意です。
また、そうすることで `user_id` を `user` と記述でき、テーブルが生成される時に _id が付与されます。
例：`|user|references|null: false, foreign_key: true|`
参考：https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

Daisekkei replied:
ご指摘ありがとうございます。
修正いたしました。

i-date reviewed (commented):
ご提出ありがとうございます！コメントいたしました^^
また、いただいたコードの追加変更差分はプルリクの[Files changed](https://github.com/Daisekkei/chat-space/pull/2/files)で確認できるのですが、groups_usersテーブルしかないようです。DB設計では、chatspaceに必要なテーブルをすべて記述するようにしましょう。

すべての修正が完了したら再度レビューを依頼しましょう。

Daisekkei committed:
Update README.md

What
groups_users以外のテーブルを追加しました。
追加テーブル
１、userテーブル
２、groupテーブル
３、commentテーブル

Why
前の変更ではgroup_userテーブルしか記載がなく、必要テーブルが不足していたため

Daisekkei reviewed (commented):
What
前の指摘箇所、３０、３１行目の
user_id => user, integer=> references
group_id => group, integer => references
への変更も済み

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+## usersテーブル
+|Column|Type|Options|
+|------|----|-------|
+|user|references|null: false, foreign_key: true|
```
usersテーブルにuser_idは不要ですのでこちらは消しましょう。

Daisekkei replied:
削除いたしました

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|------|----|-------|
+|user|references|null: false, foreign_key: true|
+|email|string|null: false|
+|user_name|integer|null: false|
```
usersテーブルのカラムなのでnameで結構です。
型はstringにしましょう。

Daisekkei replied:
カラム名の変更、型の変更いたしました

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|password|integer|null: false|
+
+### Association
+- has_many :group, through: :groups_users
```
has_manyの場合は複数形にしましょう。
他の箇所も同様です。
また、has_many :groups_usersも追加しましょう。

Daisekkei replied:
has_manyのあとに続くgroupを複数形に変更しました。
それ以外も複数形に訂正しました。
has_many: groups_usersのAssociationも追加いたしました。

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|group|references|null: false, foreign_key: true|
```
こちら1行不要ですので消しましょう。

Daisekkei replied:
groupのカラムを削除いたしました

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|group|references|null: false, foreign_key: true|
+|group_name|integer|null: false, foreign_key: true|
```
nameで結構です。型はstringにしましょう。
また、foreign_key: trueは不要です。

Daisekkei replied:
stringへの変更、foreign_key: trueを削除いたしました。

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|------|----|-------|
+|group|references|null: false, foreign_key: true|
+|group_name|integer|null: false, foreign_key: true|
+|member|integer|null: false, foreign_key: true|
```
memberは不要ですので消しましょう。

Daisekkei replied:
menberのカラムを削除いたしました。

shingoteshima replied:
こちら削除されていません。

Daisekkei replied:
menberを削除いたしました

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|member|integer|null: false, foreign_key: true|
+
+### Association
+- belongs_to :user
```
usersテーブルとのアソシエーションが間違っています。

Daisekkei replied:
1つのグループに対して、多くのユーザー が登録されているため
belongs_to :user => has_many :users
に変更いたしました

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|group|references|null: false, foreign_key: true|
+|comment|text|null: false, foreign_key: true|
```
テキスト、画像は単体で投稿可能なのでnull: falseは不要です。
また、foreign_key: trueも不要です。

Daisekkei replied:
null: falseとforeign_key: true両方とも削除いたしました。

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|------|----|-------|
+|group|references|null: false, foreign_key: true|
+|comment|text|null: false, foreign_key: true|
+|image|string|foreign_key: true|
```
foreign_key: true不要です。

Daisekkei replied:
foreign_key: trueを削除いたしました。

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|group|references|null: false, foreign_key: true|
```
`|user|references|null: false, foreign_key: true|`も追加しましょう。

Daisekkei replied:
追加いたしました。

Daisekkei committed:
Update README.md

What
1、各テーブルのIDを表すカラム箇所の削除
2、アソシエーション has_many の関連づけを複数系へ訂正
3、Groupテーブルのbelongs_to :user =>has_many :usersへ変更

Why
1、 IDはテーブル作成時に自動的に生成されるため
2、複数へ指定していなかったため
3、Groupテーブルへは複数のユーザーが関連しているため

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
+## groupテーブル
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, foreign_key: true|
```
foreign_key: trueは不要です。

Daisekkei replied:
foreign_key: trueを削除しました

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
+|member|integer|null: false, foreign_key: true|
+
+### Association
+- has_many :users
```
中間テーブルを経由しましょう。

Daisekkei replied:
中間テーブルのgroups_usersテーブルを経由するアソシエーションに変更しました

Daisekkei committed:
Update README.md

What
１、groupテーブルのnameカラムの foreign_key: trueを削除
２、groupテーブルのmemberのカラムを削除
３、userとgroupテーブルのアソシエーションに  , through: : groups_users
を追加

Why
１、groupテーブルのnameカラムには外部キー設定をしていないため
２、userテーブルとgroupテーブルとの間に中間テーブルを置いているため、groupテーブルにどのユーザーが関連されているかという情報は追加する必要がないため
３、userテーブルとgroupテーブルとの間に中間テーブルを置いているため

TOMARUMARU commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+### Association
+- has_many :groups, through: :groups_users
+- has_many :comments
+- has_many :groups, through: :groups_users
```
同じ記述が二回あります

Daisekkei replied:
二重の記述を削除しました

TOMARUMARU commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+|name|string|null: false|
+
+### Association
+- has_many :users, through: :groups_users
```
中間テーブルとも組みましょう

Daisekkei replied:
- has_many :groups_users
を追加いたしました。

Daisekkei committed:
Update README.md

What
1、userテーブルで
has_many :groups, through: :groups_users
を削除
2、user、groupテーブルのアソシエーションに
- has_many :groups_users
を追記

Why
１、has_many :groups, through: :groups_usersがダブって記載されていたため
２、中間テーブルgroups_usersと組むため

miyagit reviewed (approved):
LGTM
